azurelinuxagent/common/rdma.py

```diff
@@ -90,3 +96,133 @@ def reboot_system(self):
+        """
+        Start a thread in the background to process the RDMA tasks and returns.
+        """
+        logger.info("RDMA: starting device processing in the background.")
+        threading.Thread(target=self.process).start()
+
+    def process(self):
+        RDMADeviceHandler.update_dat_conf(dapl_config_paths, self.ipv4_addr)
+        RDMADeviceHandler.wait_rdma_device(
+            self.rdma_dev, self.device_check_timeout_sec, self.device_check_interval_sec)
+        RDMADeviceHandler.write_rdma_config_on_device(
+            self.rdma_dev, self.ipv4_addr, self.mac_addr)
+        RDMADeviceHandler.update_network_interface(self.mac_addr, self.ipv4_addr)
+
+    @staticmethod
+    def update_dat_conf(paths, ipv4_addr):
+        """
+        Looks at paths for dat.conf file and updates the ip address for the
+        infiniband interface.
+        """
+        logger.info("Updating DAPL configuration file")
+        for f in paths:
+            logger.info("RDMA: trying {0}".format(f))
+            if not os.path.isfile(f):
+                logger.info(
+                    "RDMA: DAPL config not found at {0}".format(f))
+                continue
+            logger.info("RDMA: DAPL config is at: {0}".format(f))
+            cfg = fileutil.read_file(f)
+            new_cfg = RDMADeviceHandler.replace_dat_conf_contents(
+                cfg, ipv4_addr)
+            fileutil.write_file(f, new_cfg)
+            logger.info("RDMA: DAPL configuration is updated")
+            return
+
+        raise Exception("RDMA: DAPL configuration file not found at predefined paths")
+
+    @staticmethod
+    def replace_dat_conf_contents(cfg, ipv4_addr):
+        old = "ofa-v2-ib0 u2.0 nonthreadsafe default libdaplofa.so.2 dapl.2.0 \"\S+ 0\""
+        new = "ofa-v2-ib0 u2.0 nonthreadsafe default libdaplofa.so.2 dapl.2.0 \"{0} 0\"".format(
+            ipv4_addr)
+        return re.sub(old, new, cfg)
+
+    @staticmethod
+    def write_rdma_config_on_device(path, ipv4_addr, mac_addr):
+        data = RDMADeviceHandler.get_rdma_config(ipv4_addr, mac_addr)
+        logger.info(
+            "RDMA: Updating device with configuration: {0}".format(data))
+        with open(path, "w") as c:
+            c.write(data)
+        logger.info("RDMA: Updated device with IPv4/MAC addr successfully")
+
+    @staticmethod
+    def get_rdma_config(ipv4_addr, mac_addr):
+        return 'rdmaMacAddress="{0}" rdmaIPv4Address="{1}"'.format(mac_addr, ipv4_addr)
+
+    @staticmethod
+    def wait_rdma_device(path, timeout_sec, check_interval_sec):
+        logger.info("RDMA: waiting for device {0}".format(path))
+        total_retries = timeout_sec/check_interval_sec
+        n = 0
+        while n < total_retries:
+            if os.path.exists(path):
+                logger.info("RDMA: device ready")
+                return
+            logger.info(
+                "RDMA: device not ready, sleep {0}s".format(check_interval_sec))
+            time.sleep(check_interval_sec)
```

Comment:
The effect, at least on me, when I look through logs that have the same message many times in a row is that I start to glance over things and then stuff that follows gets missed. It is exactly in the case where there is a problem that we do not want to trigger that behavior in the log reviewer. If there is a problem we'd potentially log 120 messages of "RDMA: device not ready...." If we instead would log only 3 or 4 but include the wait interval then we get the same effect, i.e. it is obvious that there is a problem, but we do not roll the log viewers eyes into the back of their head.

So something like:
RDMA: device not ready current wait time X, sleep Y

That way X starts a say 20, while Y remains at 1, this indicates that we print every 20th message and that there is something going on.